### PR TITLE
Refactor ParsedMethod, ParsedSignal and ParsedInheritedMethod

### PR DESCRIPTION
**Pull request by BenFordTytherington**
- Refactor method and inherited method to have docs passthrough
- Group common functionality between these 3 similar structs

### CODE REVIEW

**Comment by BenFordTytherington** [comment]
Is it useful to combine all these invokable types in `ParsedCxxQtData` using trait objects and enum markers? or is it better to keep them separate for preserving order.
Something like `invokables: BTreeMap<InvokableKind, Box<Vec<dyn Invokable>>` might be too complex but could also combine lots of common logic?

File: crates/cxx-qt-gen/src/parser/signals.rs [hunk 1]
```diff
@@ -34,10 +33,16 @@ pub struct ParsedSignal {
...
+    /// All the doc attributes (each line) of the signal
     pub docs: Vec<Attribute>,
 }
 
+impl Invokable for &ParsedSignal {
+    fn name(&self) -> &Name {
+        &self.name
+    }
+}
+
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-gen/src/parser/signals.rs:45]: A whole trait seems a bit overkill. Maybe prefer using a lambda in the structuring phase instead that just accesses the right member.

File: crates/cxx-qt-gen/src/generator/rust/method.rs [hunk 2]
```diff
@@ -179,7 +178,6 @@ mod tests {
             &generated.cxx_mod_contents[0],
             quote! {
                 extern "Rust" {
-                    #[doc(hidden)]
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-gen/src/generator/rust/method.rs:182]: These tests, as well as the expected output test probably still need updating.
I assume that's also why CI fails.

File: crates/cxx-qt-gen/src/parser/mod.rs [hunk 3]
```diff
@@ -25,9 +29,67 @@ use syn::{
...
+/// Struct with common fields between Invokable types.
+/// These types are ParsedSignal, ParsedMethod and ParsedInheritedMethod
+pub struct MethodFields {
+    qobject_ident: Ident,
+    mutable: bool,
+    parameters: Vec<ParsedFunctionParameter>,
+    safe: bool,
+    name: Name,
+    docs: Vec<Attribute>, // TODO: Remove this
+}
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-gen/src/parser/mod.rs:59]: Quite nice to see this being deduplicated.
I would prefer to move this type and associated functions into its own file though. You could probably also just put it into method.rs. But it doesn't really belong into the parser module.

File: crates/cxx-qt-gen/src/parser/mod.rs [hunk 4]
```diff
@@ -25,9 +29,67 @@ use syn::{
...
+
+/// Function for creating an [MethodFields] from a method and docs.
+/// These fields are shared by ParsedSignal, ParsedMethod and ParsedInheritedMethod so grouped into common logic.
+pub fn extract_common_fields(method: &ForeignItemFn, docs: Vec<Attribute>) -> Result<MethodFields> {
```
Comment (LeonMatthesKDAB) [inline crates/cxx-qt-gen/src/parser/mod.rs:63]: Why isn't this function `MethodFields::parse`?
That's the pattern we use for all other parsing.

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
The tests seem to not be up-to-date, but this is going in the right direction :)

File: crates/cxx-qt-gen/src/parser/mod.rs (hunk 4 above)
Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/parser/mod.rs]: Done

File: crates/cxx-qt-gen/src/parser/mod.rs (hunk 3 above)
Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/parser/mod.rs]: Moved into method.rs

File: crates/cxx-qt-gen/src/generator/rust/method.rs (hunk 2 above)
Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/generator/rust/method.rs:182]: Done

File: crates/cxx-qt-gen/src/parser/signals.rs (hunk 1 above)
Comment (BenFordTytherington) [reply crates/cxx-qt-gen/src/parser/signals.rs]: Done

**Review by LeonMatthesKDAB — APPROVED** [review]

**Pull request merged**